File: frontend/src/app/core/services/production.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, map } from 'rxjs';
import { environment } from '../../../environments/environment';
import { ProductionOrder, ProductionYield } from '../models/production.model';

@Injectable({ providedIn: 'root' })
export class ProductionService {
  private http = inject(HttpClient);
  private base = `${environment.apiUrl}/production`;

  listOrders(): Observable<ProductionOrder[]> {
    return this.http.get<{ data: ProductionOrder[] }>(`${this.base}/orders`).pipe(map(r => r.data));
  }

  getOrder(id: string): Observable<ProductionOrder> {
    return this.http.get<ProductionOrder>(`${this.base}/orders/${id}`);
  }

  createOrder(body: object): Observable<ProductionOrder> {
    return this.http.post<ProductionOrder>(`${this.base}/orders`, body);
  }

  updateOrder(id: string, body: object): Observable<ProductionOrder> {
    return this.http.put<ProductionOrder>(`${this.base}/orders/${id}`, body);
  }

  confirmOrder(id: string): Observable<ProductionOrder> {
    return this.http.post<ProductionOrder>(`${this.base}/orders/${id}/confirm`, {});
  }

  issueRM(id: string, body: object): Observable<ProductionOrder> {
    return this.http.post<ProductionOrder>(`${this.base}/orders/${id}/issue-rm`, body);
  }

  startProduction(id: string): Observable<ProductionOrder> {
    return this.http.post<ProductionOrder>(`${this.base}/orders/${id}/start`, {});
  }

  recordYield(id: string, body: object): Observable<ProductionYield> {
    return this.http.post<ProductionYield>(`${this.base}/orders/${id}/record-yield`, body);
  }

  completeOrder(id: string): Observable<ProductionOrder> {
    return this.http.post<ProductionOrder>(`${this.base}/orders/${id}/complete`, {});
  }

  cancelOrder(id: string): Observable<void> {
    return this.http.post<void>(`${this.base}/orders/${id}/cancel`, {});
  }

  listYields(orderId?: string): Observable<ProductionYield[]> {
    const params = orderId ? `?order_id=${orderId}` : '';
    return this.http.get<{ data: ProductionYield[] }>(`${this.base}/yields${params}`).pipe(map(r => r.data));
  }
}

File: backend/internal/service/production_service.go
package service

import (
	"context"

	"github.com/water-factory/api/internal/domain"
	"github.com/water-factory/api/internal/dto"
	"github.com/water-factory/api/internal/repository"
)

type ProductionService struct {
	repo *repository.ProductionRepo
}

func NewProductionService(repo *repository.ProductionRepo) *ProductionService {
	return &ProductionService{repo: repo}
}

func (s *ProductionService) ListOrders(ctx context.Context) ([]*domain.ProductionOrder, error) {
	return s.repo.ListOrders(ctx)
}

func (s *ProductionService) GetOrder(ctx context.Context, id string) (*domain.ProductionOrder, error) {
	return s.repo.GetOrder(ctx, id)
}

func (s *ProductionService) CreateOrder(ctx context.Context, userID string, req *dto.CreateProductionOrderRequest) (*domain.ProductionOrder, error) {
	return s.repo.CreateOrder(ctx, userID, req)
}

func (s *ProductionService) UpdateOrder(ctx context.Context, id string, req *dto.UpdateProductionOrderRequest) (*domain.ProductionOrder, error) {
	return s.repo.UpdateOrder(ctx, id, req)
}

func (s *ProductionService) ConfirmOrder(ctx context.Context, id string) (*domain.ProductionOrder, error) {
	return s.repo.ConfirmOrder(ctx, id)
}

func (s *ProductionService) IssueRM(ctx context.Context, id, userID string, req *dto.IssueRMRequest) (*domain.ProductionOrder, error) {
	return s.repo.IssueRM(ctx, id, userID, req)
}

func (s *ProductionService) StartProduction(ctx context.Context, id string) (*domain.ProductionOrder, error) {
	return s.repo.StartProduction(ctx, id)
}

func (s *ProductionService) RecordYield(ctx context.Context, id, userID string, req *dto.RecordYieldRequest) (*domain.ProductionYield, error) {
	return s.repo.RecordYield(ctx, id, userID, req)
}

func (s *ProductionService) CompleteOrder(ctx context.Context, id string) (*domain.ProductionOrder, error) {
	return s.repo.CompleteOrder(ctx, id)
}

func (s *ProductionService) CancelOrder(ctx context.Context, id string) error {
	return s.repo.CancelOrder(ctx, id)
}

func (s *ProductionService) ListYields(ctx context.Context, orderID string) ([]*domain.ProductionYield, error) {
	return s.repo.ListYields(ctx, orderID)
}
